mars/services/task/execution/ray/executor.py

```diff
@@ -152,26 +152,39 @@ def execute_subtask(
     # optimize chunk graph.
     subtask_chunk_graph = _optimize_subtask_graph(subtask_chunk_graph)
     start_chunks = _get_start_chunks(subtask_chunk_graph)
-    if isinstance(start_chunks[0].op, FetchShuffle):
-        assert len(start_chunks) == 1, start_chunks
-        # the subtask is a reducer subtask
-        n_mappers = len(inputs)
-        # some reducer may have multiple output chunks, see `PSRSshuffle._execute_reduce` and
-        # https://user-images.githubusercontent.com/12445254/168569524-f09e42a7-653a-4102-bdf0-cc1631b3168d.png
-        reducer_chunks = subtask_chunk_graph.successors(start_chunks[0])
-        reducer_operands = set(c.op for c in reducer_chunks)
-        if len(reducer_operands) != 1:
-            raise ValueError(
-                f"Subtask {subtask_id} has more than 1 reduce operands: {subtask_chunk_graph.to_dot()}"
+    input_data = {}
+    has_shuffle_chunk = False
+    input_keys = []
+    for start_chunk in start_chunks:
+        if isinstance(start_chunk.op, FetchShuffle):
```

Comment:
It's duplicated to check the `FetchShuffle` instance in `_get_start_chunks` and this iteration. Can you just check isinstance once? For example, `_get_start_chunks` returns two lists, normal fetch list and fetch shuffle list?

Reply:
Good suggestion, fixed it